### PR TITLE
[DOCS] Rewrite `terms_set` query

### PR DESCRIPTION
**Pull request by jrodewig**
## Changes
* Rewrites description of `terms_set` query
* Adds example request section
* Adds parameters sections
* Adds notes section

This is part of #40977, an effort to standardize documentation for query types.

## Before
<details>
 <summary>Before image</summary>
Before image
</details>

## After
<details>
 <summary>After image</summary>
Before image
</details>

### CODE REVIEW

File: docs/reference/query-dsl/terms-set-query.asciidoc [hunk 1]
```diff
@@ -1,121 +1,220 @@
...
-Also the `params.num_terms` parameter is available in the script to indicate the
-number of terms that have been specified.
+For an example query using the `minimum_should_match_script` parameter, see
+<<terms-set-query-script, How to use the `minimum_should_match_script`
+parameter>>.
+--
 
-An example that always limits the number of required terms to match to never
-become larger than the number of terms specified:
+`boost`::
```
Comment (cbuescher) [inline docs/reference/query-dsl/terms-set-query.asciidoc]: Like in a previous PR, maybe we could mention the "boost" parameter on all queries for completeness but link to a common paragraph for the detailed discussion.

**Review by cbuescher — COMMENTED** [review]
Thanks again @jrodewig, I didn't know this query very well and it was a pleasure reading the rewrite. Left one suggestions regarding "boost" again, also I wasn't sure about real-life use cases for this type of query so I did a bit of digging when/why this was added and found https://github.com/elastic/elasticsearch/issues/26915. There are two more or less intuitive examples where this pretty specialized query might be useful in https://github.com/elastic/elasticsearch/issues/26915#issuecomment-335100412. I don't know how far you want to take details in these reference docs but maybe there is a way of including some of these thoughts succinctly? Just an idea, maybe that makes the docs to complicated so I leave it up to you.

**Comment by jrodewig** [comment]
Thanks for the review and praise @cbuescher. 

I've updated the example and included the listed use cases with a33ee6e. Thanks for pointing out #26915. I feel like the examples are much more practical now.

I also removed the `boost` parameter documentation for now with 75e12a4.

**Review by cbuescher — APPROVED** [review]
Awesome, LGTM

**Pull request merged**